includes/admin/class-wc-rest-payments-transactions-controller.php

```diff
@@ -154,4 +166,35 @@ static function ( $filter ) {
 			}
 		);
 	}
+
+	/**
+	 * Formats the incoming transaction date as per the blog's timezone.
+	 *
+	 * @param string|null $transaction_date Transaction date to format.
+	 * @param string      $user_timezone         User's timezone passed from client.
+	 *
+	 * @return string|null The formatted transaction date as per timezone.
+	 */
+	private function format_transaction_date_with_timestamp( $transaction_date, $user_timezone ) {
+		if ( is_null( $transaction_date ) ) {
+			return $transaction_date;
+		}
+
+		// Get blog timezone
+		$blog_time = new DateTime( $transaction_date );
+		$blog_time->setTimezone( new DateTimeZone( wp_timezone_string() ) );
+
+		// Get local timezone
+		$local_time = new DateTime( $transaction_date );
+		$local_time->setTimezone( new DateTimeZone( $user_timezone ) );
+
+		// Compute time difference in minutes
+		$time_difference = ( strtotime( $blog_time->format( 'Y-m-d H:i:s' ) ) - strtotime( $local_time->format( 'Y-m-d H:i:s' ) ) ) / 60;
```

Comment:
Looks like we need to invert the condition here: `$local_time - $blog_time` so that it add or subtract automatically in the l.196 (the `+` sign should be removed)

Reply:
Good catch.